include/net/net_if.h

```diff
@@ -469,6 +469,10 @@ struct net_if {
 
 	/** Network interface instance configuration */
 	struct net_if_config config;
+
+	/** Keep track of packets pending in traffic queues */
+	int tx_pending[NET_TC_TX_COUNT];
```

Comment:
Do we care to track per-queue? 
Also, I fail to see any other usage than with CONFIG_NETWORK_POWER_MANAGEMENT  so we could avoid growing the size of the structure when this option is unset. What do you think ?

Reply:
Yes, we could go first with global counter. Also the RX tracking is not needed (I just put it there for symmetry). Noticed also a bug where the code did not check whether the net-pkt was inserted properly to the workqueue. Will send a new version shortly.

Reply:
> CONFIG_NETWORK_POWER_MANAGEMENT

BTW, should we call this CONFIG_NET_POWER_MANAGEMENT instead, to be similar like other network related config options?

Reply:
yep

Reply:
heh, actually it is already like that. Hard to focus these days...